.github/workflows/build-src.yml

```diff
@@ -68,17 +74,40 @@ jobs:
         run: |
           CCACHE_SIZE="400M"
           CACHE_DIR="/cache"
-          mkdir /output
-          BASE_OUTDIR="/output"
           BUILD_TARGET="${{ inputs.build-target }}"
           source ./ci/dash/matrix.sh
           ./ci/dash/build_src.sh
           ./ci/dash/test_unittests.sh
         shell: bash
 
+      - name: Bundle outputs and artifacts
+        id: bundle
+        run: |
+          export BUILD_TARGET="${{ inputs.build-target }}"
+          ./ci/dash/bundle-output.sh create
+          ARTIFACT_PATH="/output" ./ci/dash/bundle-build.sh
+          echo "key=build-$(head -c 512 /dev/urandom | sha256sum | head -c 64)" >> "${GITHUB_OUTPUT}"
+          echo "short-sha=$(git rev-parse --short=8 HEAD)" >> "${GITHUB_OUTPUT}"
+
+      - name: Save build outputs
+        if: |
+          inputs.build-target == 'linux64'
+          || inputs.build-target == 'linux64_nowallet'
+          || inputs.build-target == 'linux64_sqlite'
+          || inputs.build-target == 'linux64_ubsan'
+        uses: actions/cache/save@v4
+        with:
+          path: |
+            output-${{ inputs.build-target }}.tar.zst
+          key: ${{ steps.bundle.outputs.key }}
```

Comment:
why can't we use the artifacts we are uploading anyhow instead of abusing the cache system? (such as requiring delete later)

Reply:
Resolved in latest push